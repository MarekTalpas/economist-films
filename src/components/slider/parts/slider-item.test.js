// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SliderItem from './slider-item';

const dummyData = [
  {
    title: 'Episode 6 Title',
    type: 'Episode',
    id: 6,
    subtitle: 'subtitle',
    thumbnail: {
      thumb: {
        url: 'url',
      },
    },
    className: 'slider-item',
    isSelected: false,
    isBeforeSelected: false,
    isSelectedSeries: false,
  },
  {
    title: 'Series 1 Title',
    type: 'Series',
    id: 9,
    episode_count: 10,
    thumbnail: {
      thumb: {
        url: 'url',
      },
    },
    className: 'slider-item',
    isSelected: false,
    isBeforeSelected: true,
    isSelectedSeries: true,
  },
  {
    title: 'Episode 19 Title',
    type: 'Episode',
    id: 19,
    subtitle: 'subtitle',
    thumbnail: {},
    className: 'slider-item',
    isSelected: true,
    isBeforeSelected: false,
    isSelectedSeries: true,
  },
];

test('SliderItem renders correctly', () => {
  dummyData.forEach((data) => {
    const tree: string = renderer.create(<SliderItem {...data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
