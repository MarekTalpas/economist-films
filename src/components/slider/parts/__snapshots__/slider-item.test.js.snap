// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderItem renders correctly 1`] = `
<a
  className="slider-item"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="Episode 6 Title"
    className="slider-item__image"
    src="url"
  />
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Episode 6 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      subtitle
    </div>
  </div>
</a>
`;

exports[`SliderItem renders correctly 2`] = `
<a
  className="slider-item slider-item--left-offset"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="Series 1 Title"
    className="slider-item__image"
    src="url"
  />
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Series 1 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      10 episodes
    </div>
  </div>
</a>
`;

exports[`SliderItem renders correctly 3`] = `
<a
  className="slider-item slider-item--selected"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="Episode 19 Title"
    className="slider-item__image"
    src=""
  />
  <div
    className="slider-item__container"
  >
    <div
      className="slider-item__title"
    >
      Episode 19 Title
    </div>
    <div
      className="slider-item__subtitle"
    >
      subtitle
    </div>
  </div>
</a>
`;
